### ISSUE TITLE
Could TextAttack only attack substring of input.

### ISSUE THREAD

**Issue by ludybupt** [QData/TextAttack#733]
I would  like to employ TextAttack  for further attack.
does TextAttack  support only setting attack range of input?

**Comment by jxmorris12** [comment]
Yes, definitely. You can implement a PreTransformationConstraint that filters out transformations to the words that you don't want to alter. Here's an example, which we used to prevent modifying the premise (or hypothesis) in SNLI: https://github.com/QData/TextAttack/blob/master/textattack/constraints/pre_transformation/input_column_modification.py#L11

**Comment by qiyanjun** [comment]
Answered above (use a `PreTransformationConstraint`, e.g. `InputColumnModification`) with no follow-up needed. Closing — feel free to reopen if that doesn't cover your case.
